aten/src/ATen/TensorIndexing.h

```diff
@@ -198,16 +198,22 @@ static inline Tensor applySlice(
     int64_t step,
     bool disable_slice_optimization,
     const at::Device& self_device,
-    const IntArrayRef& self_sizes) {
+    const c10::optional<IntArrayRef> & self_sizes) {
   // TODO: implement negative step
   TORCH_CHECK_VALUE(step > 0, "step must be greater than zero");
 
-  // Skip this optimization if we are tracing, as the trace may be polymorphic
-  // over the shape of the `self` tensor, and we still want to record
-  // the slice.
-  int64_t length = (self_device == at::kCPU || self_device == at::kCUDA) ? self_sizes[dim] : self.size(dim);
-  if (!disable_slice_optimization && start == 0 && stop == length && step == 1) {
-    return self;
+  // TODO: nested tensor does not have a size (yet) so for now we represent its size as null
```

Comment:
No need to duplicate this comment I think.
we only want a single copy of it at the top level.